net/ser2net/files/ser2net.init

```diff
@@ -21,34 +21,41 @@ list_cb_append() {
 ser2net_default() {
 	local cfg="$1"
 	local key val
+	local baudrate parity databits stopbits
 
 	config_get baudrate "$cfg" speed
-	config_get parity "$cfg" parity
-	parity=`echo "$parity" | head -c 1`
-	[ -z "$parity" ] && parity=n
+	if [ -n "$baudrate" ]; then
+		config_get parity "$cfg" parity
+		case "$parity" in
+			[Nn]one) parity=n ;;
+			[Oo]dd) parity=o ;;
+			[Ee]ven) parity=e ;;
+			"") ;;
+			*) return 1
+		esac
 
-	config_get databits "$cfg" databits
-	if [ -n "$databits" ]; then
-		[ "$databits" -lt 5 -o "$databits" -gt 8 ] && return 1
+		if [ -n "$parity" ]; then
+			config_get databits "$cfg" databits 8
+			[ "$databits" -ge 5 ] && [ "$databits" -le 9 ] || return 1
+
+			config_get stopbits "$cfg" stopbits 1
+			case "$stopbits" in
+				1) ;;
+				2) ;;
+				*) return 1
+			esac
+		fi
+
+		echo "default:"
+		echo "  name: speed"
+		echo "  value: $baudrate${parity:+$parity$databits$stopbits}"
 	fi
 
-	config_get stopbits "$cfg" stopbits
-	case "$stopbits" in
-		1) ;;
-		2) ;;
-		"") ;;
-		*) return 1
-	esac
-
-	echo "default:"
-	echo "  name: speed"
-	echo "  value: $baudrate$parity$databits$stopbits"
-
 	for key in chardelay_scale chardelay_min; do
 		config_get val "$cfg" "$key"
 		[ -n "$val" ] || continue
 		key=`echo "$key" | tr '_' '-'`
-		echo "default: &$cfg_$key"
+		echo "default:"
```

Comment:
I'm a bit nervous about removing the &$cfg_$key anchor.
Strictl yaml doesn't allow duplicate keys and I once took down a prod environment when the yaml parser overwrote the previous values.
Should be OK Just test that duplicate keys works as expected I guess.

Reply:
To be honest, I just had a look at the shipped template of ser2net and there are no anchors - this is why I thought that they are not strictly required. But I have no strong opinion - we can also leave it in if you prefer.

Reply:
You are likely right. I'm just cautious from a mistake I made 10+ years ago.
The yaml spec does require unique non-duplicate keys.
And alias/anchors shouldn't matter if they are duplicate.
The ser2net repo [ser2net.yaml](https://github.com/cminyard/ser2net/blob/master/ser2net.yaml
) uses anchors for all blocks except `default:`. 

Anyhow, ser2net uses [libyaml](https://pyyaml.org/wiki/LibYAML) which is a stream parser. It's likely only an issue if parsing into memory and then accessing memory by key...

Nonetheless I'll quickly tag @cminyard for his comment. Tho maybe not worth making any fuss over.